tests/flow/test_comprehension_functions.py

```
@@ -0,0 +1,454 @@
+
+    def test09_nested_list_comprehension(self):
+        query = """RETURN [elem IN [nested_val IN range(0, 6) WHERE nested_val % 2 = 0] WHERE elem * 2 >= 4 | elem * 2]"""
+        actual_result = self.graph.query(query)
+        expected_result = [[[4, 8, 12]]]
+        self.env.assertEquals(actual_result.result_set, expected_result)
+
+    def test10_any_all_comprehension_acceptance(self):
+        # Reject ANY and ALL comprehensions that don't include a WHERE predicate.
+        try:
+            self.graph.query("RETURN any(x IN [1,2])")
+            assert(False)
```

Ruff (0.12.2)

Do not `assert False` (`python -O` removes these calls), raise `AssertionError()`

Replace `assert False`

(B011)